src/main/scala/beam/agentsim/agents/ridehail/RideHailAgent.scala

```diff
@@ -265,8 +343,10 @@ class RideHailAgent(
       )
       eventsManager.processEvent(new PersonEntersVehicleEvent(tick, Id.createPersonId(id), vehicle.id))
       val isTimeForShift = shifts.isEmpty || shifts.get
-        .exists(shift => shift.lowerBound <= tick && shift.upperBound >= tick)
+        .find(shift => shift.range.lowerBound <= tick && shift.range.upperBound >= tick)
```

Comment:
I don't see why this was changed to use the less efficient version - it should remain an exists.